File: internal/integration/postgresql/client.test.ts
import { test, expect, beforeAll } from "vitest";
import { newClient, QXClient } from "./db";

let c: QXClient;

beforeAll(async () => {
  c = newClient();
});

test("string array", async () => {
  let strings = ["test1", "test2"];
  let user = await c.queryUser().create({ strings });
  expect(user.strings).toEqual(strings);
  let row = await c.queryUser().find(user.id);
  expect(row.strings).toEqual(strings);
});

test("text array", async () => {
  let texts = ["test1", "test2"];
  let user = await c.queryUser().create({ texts });
  expect(user.texts).toEqual(texts);
  let row = await c.queryUser().find(user.id);
  expect(row.texts).toEqual(texts);
});

test("integer array", async () => {
  let integers = [1, 2];
  let user = await c.queryUser().create({ integers });
  expect(user.integers).toEqual(integers);
  let row = await c.queryUser().find(user.id);
  expect(row.integers).toEqual(integers);
});

File: internal/integration/postgresql/client_test.go
package main

import (
	"log"
	"testing"

	"github.com/stretchr/testify/require"
	"github.com/swiftcarrot/queryx/internal/integration/postgresql/db"
)

var c *db.QXClient

func init() {
	client, err := db.NewClient()
	if err != nil {
		log.Fatal(err)
	}
	c = client
}

func TestStringArray(t *testing.T) {
	strings := []string{"test1", "test2"}
	user, err := c.QueryUser().Create(c.ChangeUser().SetStrings(strings))
	require.NoError(t, err)
	require.Equal(t, strings, user.Strings)
	row, err := c.QueryUser().Find(user.ID)
	require.NoError(t, err)
	require.Equal(t, strings, row.Strings)
}

func TestTextArray(t *testing.T) {
	texts := []string{"test1", "test2"}
	user, err := c.QueryUser().Create(c.ChangeUser().SetTexts(texts))
	require.NoError(t, err)
	require.Equal(t, texts, user.Texts)
	row, err := c.QueryUser().Find(user.ID)
	require.NoError(t, err)
	require.Equal(t, texts, row.Texts)
}

func TestIntegerArray(t *testing.T) {
	integers := []int{1, 2}
	user, err := c.QueryUser().Create(c.ChangeUser().SetIntegers(integers))
	require.NoError(t, err)
	require.Equal(t, integers, user.Integers)
	row, err := c.QueryUser().Find(user.ID)
	require.NoError(t, err)
	require.Equal(t, integers, row.Integers)
}
